glusterd2/commands/volumes/volume-create.go

```diff
@@ -47,7 +47,11 @@ func validateVolCreateReq(req *api.VolCreateReq) error {
 		return errors.New("invalid Volume Size, Minimum size required is " + strconv.Itoa(minVolumeSize))
 	}
 
-	if req.Size == 0 && len(req.Subvols) <= 0 {
+	if req.SelfProvision && req.Size == 0 {
+		return errors.New("volume size should provide to perform auto create")
+	}
+
+	if !req.SelfProvision && len(req.Subvols) <= 0 {
```

Comment:
Since you requested for smartvol, given bricks should omit or we should have the capability of having both. But I will let @aravindavk answer the question as it is not specific to this PR.